dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/EnrollmentPersister.java

```diff
@@ -126,6 +142,7 @@ protected TrackerSideEffectDataBundle handleSideEffects(
         .accessedBy(bundle.getUsername())
         .enrollment(enrollment)
         .program(enrollment.getProgram())
+        .triggerEvent(bundle.getPreheat().getTriggers())
```

Comment:
Wouldn't be easier to do everything in `handleSideEffects` method?

You are adding a new abstract method, to add data to the Preheat and then read those data in the `handleSideEffects` method. If you calculate the triggers here, there is no need to add the triggers in the Preheat and to have a new abstract method

Reply:
yes i did think about it. Problem is `handleSideEffect` is called after `updatePreheat` which makes `isNew` method to returns incorrect result. So either we need to have this separate method or we need to move `handleSideEffect` before `updatePreheat`.

Reply:
ok